modules/redpanda/admin_api.go

```diff
@@ -11,8 +11,10 @@ import (
 )
 
 type AdminAPIClient struct {
-	BaseURL string
-	client  *http.Client
+	BaseURL  string
+	Username string
+	Password string
```

Comment:
last-minute question (sorry): do we need these fields to be exported? If not, then let's make them private and expose `Username() string` and `Password() string` getter functions. Wdyt?

Reply:
Don't think we need the getters, but yer thats a good direction as it reduces the API surface area.

Reply:
I have made the changes to private for the admin api client properties. In a sense it's a breaking change but clients of the module if they were using this `AdminAPIClient` struct.
I have also added accessors but can remove them too. I don't think we really need them.

Reply:
If we don't change the `BaseURL` then I do not see the breaking change, please correct me if I'm wrong. I meant the new fields for User and Password in isolation.

Reply:
I have changed base URL back to public `BaseURL`.